client/my-sites/email/email-providers-comparison/index.jsx

```diff
@@ -365,27 +374,28 @@ class EmailProvidersComparison extends React.Component {
 					comment: '%(googleMailService)s can be either "G Suite" or "Google Workspace"',
 			  } );
 
-		const formFields = domain ? (
-			<FormFieldset>
-				<GSuiteNewUserList
-					extraValidation={ identityMap }
-					domains={ [ domain ] }
-					onUsersChange={ this.onGoogleUsersChange }
-					selectedDomainName={ selectedDomainName }
-					users={ googleUsers }
-					onReturnKeyPress={ this.onGoogleFormReturnKeyPress }
-				>
-					<Button
-						className="email-providers-comparison__gsuite-user-list-action-continue"
-						primary
-						busy={ this.state.addingToCart }
-						onClick={ this.onGoogleConfirmNewUsers }
+		const formFields =
+			cartDomainName || domain ? (
+				<FormFieldset>
+					<GSuiteNewUserList
+						extraValidation={ identityMap }
+						domains={ [ cartDomainName || domain ] }
+						onUsersChange={ this.onGoogleUsersChange }
+						selectedDomainName={ selectedDomainName }
+						users={ googleUsers }
+						onReturnKeyPress={ this.onGoogleFormReturnKeyPress }
```

Comment:
This might be simpler to understand/follow if we build the domain list separately, especially because we should not be passing the `domains` prop when we don't have a domain object.
When we have `domain`, we should pass in `[ domain ]`, but when we don't have `domain` we should leave the prop `null`. Take a look at this line of code: https://github.com/Automattic/wp-calypso/blob/49ea2453c8d150e169d841b105ccd23a69e9a0b6/client/components/gsuite/gsuite-new-user-list/index.tsx#L76

```suggestion
		const domainList = domain ? [ domain ] : null;
		const formFields =
			hasCartDomain || domain ? (
				<FormFieldset>
					<GSuiteNewUserList
						extraValidation={ identityMap }
						domains={ domainList }
						onUsersChange={ this.onGoogleUsersChange }
						selectedDomainName={ selectedDomainName }
						users={ googleUsers }
						onReturnKeyPress={ this.onGoogleFormReturnKeyPress }
```